event/nattype.go

```diff
@@ -0,0 +1,19 @@
+package event
+
+import "github.com/libp2p/go-libp2p-core/network"
+
+// NATDeviceProtocol is the transport protocol for which the NAT Device Type has been determined.
+type NATDeviceProtocol int
+
+const (
+	// NATDeviceUDP means that the NAT Device Type has been determined for the UDP Protocol.
+	NATDeviceUDP NATDeviceProtocol = iota
+	// NATDeviceTCP means that the NAT Device Type has been determined for the TCP Protocol.
+	NATDeviceTCP
+)
```

Comment:
Why is this in package `event`, but `NATDeviceType` is in `network`?

Reply:
Because `event. NATDeviceType ` dosen't make sense. We might have references to just the NATDeviceType in the code elsewhere and storing it as `network.NATDeviceType` is more readable.

We do something similar with `network.Reachability`.

Reply:
That makes sense. But what about `network.NATTransportProtocol`?
Not sure about our package structure here tbh, just asking.

Reply:
@marten-seemann 

On further thought, I agree with you. Have fixed this by moving the Transport Protocol to the `network` package as well.